src/composables/usePagination.ts

```diff
@@ -0,0 +1,48 @@
+import { type Ref, ref, computed, watch } from 'vue';
+
+export function usePagination(totalPages: Ref<number>, initialItemsPerPage: number) {
+  const minItemsPerPage = 5;
+  const maxItemsPerPage = 20;
+  const maxPagesToShow = 7;
```

Comment:
I would change this to be configurable but optional with default values
```suggestion
interface PaginationOptions {
  minItemsPerPage?: number,
  maxItemsPerPage?: number,
  maxPagesToShow?: number,
}

const defaultPaginationOptions: PaginationOptions = {
  minItemsPerPage: 5,
  maxItemsPerPage: 20,
  maxPagesToShow: 7,
}

export function usePagination(totalPages: Ref<number>, initialItemsPerPage: number, options?: PaginationOptions) {
  const {  minItemsPerPage,maxItemsPerPage, maxPagesToShow } = {...defaultPaginationOptions, ...options}
```

Using this patterns allows the user to set only part of the options and keep the default for the rest.
Since this is now part of the input, can be removed from the `return`

Reply:
Checking this point I noticed min and max items per page are not used in the `usePagination` but on the components, as is needed only for the condition before updating the setting.

Because of that I considered removing them from the composable, and adjusting the `maxPagesToShow` to be optional parameter with default value 7.